### PR TITLE
executor: add step handler errors

### PR DESCRIPTION
**Pull request by cmatsuoka**
Implement the error classes used during step handling.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by cjp256 — APPROVED** [review]

File: craft_parts/errors.py [hunk 1]
```diff
@@ -305,3 +305,78 @@ def __init__(
...
+    """
+
+    def __init__(self, *, part_name: str, scriptlet_name: str, exit_code: int):
+        self.part_name = part_name
+        self.scriptlet_name = scriptlet_name
+        self.exit_code = exit_code
+        brief = (
+            f"{scriptlet_name!r} in part {part_name!r} failed with code {exit_code}."
+        )
+        resolution = "Review the scriptlet and make sure it's correct."
```
Comment (sergiusens) [inline craft_parts/errors.py:381]:
```suggestion
        resolution = "Review the scriptlet and make sure it is correct."
```

File: craft_parts/errors.py [hunk 2]
```diff
@@ -305,3 +305,78 @@ def __init__(
...
+
+    def __init__(self, *, part_name: str, scriptlet_name: str, message: str):
+        self.part_name = part_name
+        self.scriptlet_name = scriptlet_name
+        self.message = message
+        brief = (
+            f"{scriptlet_name!r} in part {part_name!r} executed an invalid control "
+            f"API call: {message}."
+        )
+        resolution = "Review the scriptlet and make sure it's correct."
```
Comment (sergiusens) [inline craft_parts/errors.py:361]:
```suggestion
        resolution = "Review the scriptlet and make sure it is correct."
```

**Review by sergiusens — APPROVED** [review]
Just comments about contractions

File: craft_parts/errors.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_parts/errors.py:381]: There are many instances of "it's" in error messages. I'll fix all of them in a single pass, in a different PR.

File: craft_parts/errors.py (hunk 2 above)
Comment (cmatsuoka) [reply craft_parts/errors.py:361]: To be fixed in a later PR.

**Pull request merged**